### PR TITLE
FIX: always show filtered site settings after navigation

### PR DESCRIPTION
**Pull request by xrav3nz**
This PR fixes a problem that, after filtering the site settings:

If you navigate away by, for example, going to the "Customize" tab, and come back to the "Settings" tab:

You end up in a weird state that the filter is set, but the site settings aren't filtered.

### CODE REVIEW

File: app/assets/javascripts/admin/routes/admin-site-settings.js.es6 [hunk 1]
```diff
@@ -10,9 +10,10 @@ export default Discourse.Route.extend({
...
   afterModel(siteSettings) {
-    this.controllerFor("adminSiteSettings").set(
-      "allSiteSettings",
-      siteSettings
-    );
+    const controller = this.controllerFor("adminSiteSettings");
+
+    if (!controller.get("visible")) {
+      controller.set("visible", siteSettings);
+    }
```
Comment (xrav3nz) [inline app/assets/javascripts/admin/routes/admin-site-settings.js.es6]: I really want to move this to the constructor in `controllers/admin-site-settings.js.es6`, but then  `visible` wouldn't be initialized in time for `routes/admin-site-settings-index.js.es6`:

https://github.com/discourse/discourse/blob/e10236cb6863777c4eb96765b5fb70e1965afa3f/app/assets/javascripts/admin/routes/admin-site-settings-index.js.es6#L9

File: app/assets/javascripts/admin/controllers/admin-site-settings.js.es6 [hunk 2]
```diff
@@ -2,6 +2,8 @@ import debounce from "discourse/lib/debounce";
 
 export default Ember.Controller.extend({
   filter: null,
+  allSiteSettings: Ember.computed.alias("model"),
+  visible: null,
```
Comment (ZogStriP) [inline app/assets/javascripts/admin/controllers/admin-site-settings.js.es6]: To me `visible` is a boolean, not a list of visible settings. Maybe call it `visibleSiteSettings`?

Comment (xrav3nz) [reply app/assets/javascripts/admin/controllers/admin-site-settings.js.es6]: 👍 done

**Comment by SamSaffron** [comment]
yay, nice fix!

**Pull request merged**